addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactP20Handler.java

```diff
@@ -0,0 +1,967 @@
+
+		logger.debug("Handler disposed.");
+	}
+
+	protected void configureListener(int listenerPort) {
+
+		// open the listener port
+		try {
+			listenerChannel = DatagramChannel.open();
+			listenerChannel.socket().bind(new InetSocketAddress(listenerPort));
+			listenerChannel.configureBlocking(false);
+
+			logger.info("Listening for incoming data on {}",
+					listenerChannel.getLocalAddress());
+
+			synchronized (selector) {
+				selector.wakeup();
+				try {
+					listenerKey = listenerChannel.register(selector,
+							listenerChannel.validOps());
+
+					if (getThing().getStatus() == ThingStatus.OFFLINE) {
+						updateStatus(ThingStatus.ONLINE);
+					}
+
+				} catch (ClosedChannelException e1) {
+					logger.error(
+							"An exception occurred while registering a selector: {}",
+							e1.getMessage());
+				}
+			}
+		} catch (Exception e2) {
+			logger.error(
+					"An exception occurred while creating the Listener Channel on port number {} ({})",
+					listenerPort, e2.getMessage());
+		}
+	}
+
+	protected ByteBuffer onReadable(DatagramChannel theChannel, int bufferSize,
+			InetAddress permittedClientAddress) {
+		lock.lock();
+		try {
+
+			SelectionKey theSelectionKey = theChannel.keyFor(selector);
+
+			if (theSelectionKey != null) {
+
+				synchronized (selector) {
+					try {
+						selector.selectNow();
+					} catch (IOException e) {
+						logger.error(
+								"An exception occurred while selecting: {}",
+								e.getMessage());
+					}
+				}
+
+				Iterator<SelectionKey> it = selector.selectedKeys().iterator();
+				while (it.hasNext()) {
+					SelectionKey selKey = (SelectionKey) it.next();
+					it.remove();
+					if (selKey.isValid() && selKey.isReadable()
+							&& selKey == theSelectionKey) {
+
+						ByteBuffer readBuffer = ByteBuffer.allocate(bufferSize);
+						int numberBytesRead = 0;
+						boolean error = false;
+
+						if (selKey == listenerKey) {
+							try {
+								InetSocketAddress clientAddress = (InetSocketAddress) theChannel
+										.receive(readBuffer);
+								if (clientAddress.getAddress().equals(
+										permittedClientAddress)) {
+									logger.trace(
+											"Received {} on the listener port from {}",
+											new String(readBuffer.array()),
+											clientAddress);
+									numberBytesRead = readBuffer.position();
+								} else {
+									logger.warn(
+											"Received data from '{}' which is not the permitted remote address '{}'",
+											clientAddress,
+											permittedClientAddress);
+									return null;
+								}
+							} catch (Exception e) {
+								error = true;
+							}
+
+						} else {
+
+							try {
+								numberBytesRead = theChannel.read(readBuffer);
+							} catch (NotYetConnectedException e) {
+								logger.warn(
+										"The channel '{}' is not yet connected: {}",
+										theChannel, e.getMessage());
+								error = true;
+							} catch (IOException e) {
+								// If some other I/O error occurs
+								logger.warn(
```

Comment:
this occurs on a normal timeout - so either remove the log or reduce it to debug